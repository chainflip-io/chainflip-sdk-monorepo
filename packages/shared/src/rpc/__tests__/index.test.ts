import axios from 'axios';
import {
  fundingEnvironment,
  ingressEgressEnvironment,
  poolsEnvironment,
  swappingEnvironment,
} from '../../tests/fixtures';
import {
  getFundingEnvironment,
  getSwappingEnvironment,
  getIngressEgressEnvironment,
  getPoolsEnvironment,
} from '../index';

jest.mock('axios');

const mockResponse = (data: any) =>
  jest.mocked(axios.post).mockResolvedValueOnce({ data });

describe('getFundingEnvironment', () => {
  it('retrieves the funding environment', async () => {
    const spy = mockResponse(fundingEnvironment());

    expect(await getFundingEnvironment({ network: 'perseverance' })).toEqual({
      redemptionTax: 0x4563918244f40000n,
      minimumFundingAmount: 0x8ac7230489e80000n,
    });
    expect(spy.mock.calls).toMatchSnapshot();
  });
});

describe('getSwappingEnvironment', () => {
  it('retrieves the swapping environment', async () => {
    const spy = mockResponse(swappingEnvironment('0x4563918244f40000'));

    expect(await getSwappingEnvironment({ network: 'perseverance' })).toEqual({
      minimumSwapAmounts: {
        Bitcoin: { BTC: 0x4563918244f40000n },
        Ethereum: {
          ETH: 0x4563918244f40000n,
          USDC: 0x4563918244f40000n,
          FLIP: 0x4563918244f40000n,
        },
        Polkadot: { DOT: 0x4563918244f40000n },
      },
      maximumSwapAmounts: {
        Bitcoin: {
          BTC: 10000000000000000000n,
        },
        Ethereum: {
          ETH: null,
          FLIP: null,
          USDC: 10000000000000000000n,
        },
        Polkadot: {
          DOT: null,
        },
      },
    });
    expect(spy.mock.calls).toMatchSnapshot();
  });
});

describe('getIngressEgressEnvironment', () => {
  it('retrieves the ingress egress environment', async () => {
    const spy = mockResponse(ingressEgressEnvironment('0x4563918244f40000'));

    expect(
      await getIngressEgressEnvironment({ network: 'perseverance' }),
    ).toEqual({
      minimumDepositAmounts: {
        Bitcoin: { BTC: 0x4563918244f40000n },
        Ethereum: {
          ETH: 0x4563918244f40000n,
          USDC: 0x4563918244f40000n,
          FLIP: 0x4563918244f40000n,
        },
        Polkadot: { DOT: 0x4563918244f40000n },
      },
    });
    expect(spy.mock.calls).toMatchSnapshot();
  });
});

describe('getPoolsEnvironment', () => {
  it('retrieves the pools environment', async () => {
    const spy = mockResponse(poolsEnvironment());

    expect(
      await getPoolsEnvironment({ network: 'perseverance' }),
    ).toMatchSnapshot('pool environment');
    expect(spy.mock.calls).toMatchSnapshot();
  });
});
