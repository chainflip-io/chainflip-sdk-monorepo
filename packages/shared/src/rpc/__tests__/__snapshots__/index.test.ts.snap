// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllBoostPoolsDepth retrieves the boost pools depth balance 1`] = `
[
  [
    "https://archive.perseverance.chainflip.io",
    [
      {
        "id": "1",
        "jsonrpc": "2.0",
        "method": "cf_boost_pools_depth",
        "params": [],
      },
    ],
  ],
]
`;

exports[`getAllBoostPoolsDepth retrieves the boost pools depth balance: boost pools depth 1`] = `
[
  {
    "asset": "ArbEth",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "ArbEth",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "ArbEth",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "ArbUsdc",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "ArbUsdc",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "ArbUsdc",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "Btc",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "Btc",
    "availableAmount": 100010n,
    "tier": 10,
  },
  {
    "asset": "Btc",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "Eth",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "Eth",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "Eth",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "Flip",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "Flip",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "Flip",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "Usdc",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "Usdc",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "Usdc",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "Usdt",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "Usdt",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "Usdt",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "Dot",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "Dot",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "Dot",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "Sol",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "Sol",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "Sol",
    "availableAmount": 0n,
    "tier": 30,
  },
  {
    "asset": "SolUsdc",
    "availableAmount": 0n,
    "tier": 5,
  },
  {
    "asset": "SolUsdc",
    "availableAmount": 0n,
    "tier": 10,
  },
  {
    "asset": "SolUsdc",
    "availableAmount": 0n,
    "tier": 30,
  },
]
`;

exports[`getFundingEnvironment retrieves the funding environment 1`] = `
[
  [
    "https://archive.perseverance.chainflip.io",
    [
      {
        "id": "1",
        "jsonrpc": "2.0",
        "method": "cf_funding_environment",
        "params": [],
      },
    ],
  ],
]
`;

exports[`getIngressEgressEnvironment retrieves the ingress egress environment 1`] = `
[
  [
    "https://archive.perseverance.chainflip.io",
    [
      {
        "id": "1",
        "jsonrpc": "2.0",
        "method": "cf_ingress_egress_environment",
        "params": [],
      },
    ],
  ],
]
`;

exports[`getSwappingEnvironment retrieves the swapping environment 1`] = `
[
  [
    "https://archive.perseverance.chainflip.io",
    [
      {
        "id": "1",
        "jsonrpc": "2.0",
        "method": "cf_swapping_environment",
        "params": [],
      },
    ],
  ],
]
`;
